datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -80,7 +79,7 @@ enum EvalMethod {
 ///     [WHEN ...]
 ///     [ELSE result]
 /// END
-#[derive(Debug, Hash)]
+#[derive(Debug, Hash, PartialEq, Eq)]
```

Comment:
How can it permit auto derivation of `PartialEq` for `CaseExpr` 🤔 -- we need manual impl's for others

Reply:
Yeah, this is probably the strangest part of the bug. If you have double wrappers around `dyn Trait` then there is no issue: https://github.com/rust-lang/rust/issues/78808#issuecomment-1664012270
`CaseExpr` doesn't have any `dyn Trait` fields with a single wrapper so the `derive` macro has no problem.

Reply:
Should we create a ticket to remove those impl's once the bug is resolved?

Reply:
How about https://github.com/apache/datafusion/issues/13196?